packages/rpc-methods/src/restricted/dialog.ts

```diff
@@ -0,0 +1,298 @@
+import {
+  PermissionSpecificationBuilder,
+  PermissionType,
+  RestrictedMethodOptions,
+  ValidPermissionSpecification,
+} from '@metamask/controllers';
+import { assertExhaustive } from '@metamask/snap-utils';
+import { hasProperty, isObject, NonEmptyArray } from '@metamask/utils';
+import { ethErrors } from 'eth-rpc-errors';
+
+const methodName = 'snap_dialog';
+
+export enum DialogType {
+  alert = 'alert',
+  confirmation = 'confirmation',
+  prompt = 'prompt',
+}
+
+export type AlertFields = {
```

Comment:
Could we merge some of these types? There is significant overlap.

Reply:
I liked it less with the fields merged, but feel free to do whatever makes sense.

Reply:
I'm fine with this as well.